HC-53 Add custom emoji support

neelr opened the pull request:
Add emoji support to scrapbook﻿

sampoder commented on components/emoji.js:
```diff
@@ -0,0 +1,30 @@
+                    }
+                    setImage("https://emoji.slack-edge.com/T0266FRGM/parrot/c9f4fddc5e03d762.gif")
+                })
+        } catch (e) { alert("POOPIE") }
```
lol wut, sounds super cool!

neelr replied:
Oof lol that was for error checking, I'll probably delete that

lachlanjc commented on components/emoji.js:
```diff
@@ -0,0 +1,30 @@
+import { memo, useState, useEffect } from 'react'
+import { LazyLoadImage } from 'react-lazy-load-image-component'
+
+const Emoji = ({ name }) => {
```
Let’s memoize this component. You’re already importing it :)

lachlanjc commented on components/emoji.js:
```diff
@@ -0,0 +1,30 @@
+    let [image, setImage] = useState()
+    useEffect(() => {
+        try {
+            fetch("https://emoji-ranker.now.sh/custompng")
```
This approach means that every custom emoji will make a network request to the same endpoint. For mentions, that works because 1) they’re memoized 2) we have to hit a different endpoint for each username, but here, it’s pulling the same list. The easiest way to fix that I think would be setting a cache header on the API response, but if that doesn’t work, perhaps moving it to static props makes sense?

lachlanjc commented:
Hey! Appreciate the work on the Gala page, but can you keep that in a separate PR so we can merge when each is ready?

neelr commented:
Great (sorry just saw those today), I removed the gala stuff, and memoized the emoji component! Also on the emoji ranker backend I added cache control to 10 minutes since it shouldn't change too much.

lachlanjc commented:
Thanks! I’m merging into another branch so I can make some changes since we’ve changed image loading etc since you made this.
